manifests/osfamily/windows.pp

```diff
@@ -0,0 +1,37 @@
+class puppet_agent::osfamily::windows {
+
+  $_arch = $::kernelmajversion ?{
+    /^5\.\d+/ => 'x86', # x64 is never allowed on windows 2003
+    default   => $::puppet_agent::arch,
+  }
+
+  $_source = $::puppet_agent::source ? {
+    undef   => "https://downloads.puppetlabs.com/windows/puppet-agent-${_arch}-latest.msi",
+    default => $::puppet_agent::source,
+  }
+
+  $_msi_location = $_source ? {
+    /^puppet:/ => "${tmpdir}\\puppet-agent.msi",
+    default    => $_source,
+  }
+
+  if $_source =~ /^puppet:/ {
```

Comment:
Do we want to avoid copying the msi if it's on a network share? It might make sense to always copy it locally (installs over network have always given me trouble). Or does msiexec take care of that?